### DISCUSSION TITLE
Rust?

### DISCUSSION THREAD

**Discussion by Nubro01** [Tribler/tribler#7270] [General]
Python is a slow and carbage collecting monster and I am wondering if certain parts of Tribler could be recreated in Rust?

**Comment by hbiyik** [comment]
I think the answer is yes, but the real question who is goint to do that with the limited resources :)

**Reply by synctext** [reply]
indeed, with 17 years of effort now by 10-20 people. Big hobby project to port to Rust. Even partially. Any volunteers? :-)

**Reply by hbiyik** [reply]
> any volunteers? :-)

Assuming there would be volunteers to port who is going to maintain is even harder task. OSS without coorporate interest is unfortunetely is a fairy tale in todays industry.

**Reply by nitmeare** [reply]
Maybe ChatGPT can help with that. I found it quite usefull to translate code from one programming language into an other.

**Reply by Solomon1732** [reply]
For the near future, ChatGPT and other LLMs will output code with non-expert quality at best. At least in my opinion. Tribler is not a trivial project in terms of design and whitepaper alone, not to mention security and other worries.

If anyone reimplements Tribler in any language using ChatGPT for the heavy lifting, then I will not be using it.

**Reply by qstokkink** [reply]
@Solomon1732 I can assert that none of the Tribler devs have used LLMs for any code. Everything is lovingly handcrafted.

Full transparency: the only two LLM-generated things on this repo are the **release messages** for ~~Tribler v7.14.0 and Tribler v7.14.0-RC1~~ _CORRECTION_: actually the [Tribler v7.13.0](https://github.com/Tribler/tribler/releases/tag/v7.13.0) messages were.

**Reply by Solomon1732** [reply]
@qstokkink yes, I know. I replied to @nitmeare 🙂
